recipes/OCR_pre_annotations.ipynb

```diff
@@ -0,0 +1,368 @@
+    "  - activate the API for your project (You might need to associate facturation information if you haven't already)\n",
+    "  \n",
+    "Now that the API is activated we will need to get an API in order to call later in our project:\n",
+    "  - go to \"API and services\"/indentification\n",
+    "  - create a service account with authorization to use the vision API\n",
+    "  \n",
+    "On the service account details page:\n",
+    "  - click on add a key\n",
+    "  - download the key using json format\n",
+    "  - place the key in the folder of the project\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Install google cloud api using: `pip install --upgrade google-cloud-storage`\n",
+    "\n",
+    "We can now start to code to add ocr annotations to the asset metadata !"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Declare the path to your API_KEY\n",
+    "PATH_API_KEY ="
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def implicit():\n",
+    "    from google.cloud import storage\n",
+    "\n",
+    "    # If you don't specify credentials when constructing the client, the\n",
+    "    # client library will look for credentials in the environment.\n",
+    "    storage_client = storage.Client()\n",
+    "\n",
+    "    # Make an authenticated API request\n",
+    "    buckets = list(storage_client.list_buckets())\n",
+    "    print(buckets)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def detect_text(path):\n",
+    "    \"\"\"Detects text in the file.\"\"\"\n",
+    "    from google.cloud import vision\n",
+    "    import io\n",
+    "    credentials = service_account.Credentials.from_service_account_file(PATH_API_KEY)\n",
+    "    client = vision.ImageAnnotatorClient(credentials=credentials)\n",
+    "\n",
+    "    with io.open(path, 'rb') as image_file:\n",
+    "        content = image_file.read()\n",
+    "\n",
+    "    image = vision.types.Image(content=content)\n",
+    "\n",
+    "    response = client.text_detection(image=image)\n",
+    "    texts = response.text_annotations\n",
+    "    textAnnotations = []\n",
+    "\n",
+    "    for text in texts:\n",
+    "        \n",
+    "        vertices = [{\"x\": vertex.x, \"y\": vertex.y}\n",
+    "                    for vertex in text.bounding_poly.vertices]\n",
+    "        \n",
+    "        tmp = {\"description\": text.description,\n",
+    "               \"boundingPoly\": {\n",
+    "                      \"vertices\": vertices,\n",
+    "                  },\n",
+    "              }\n",
+    "        \n",
+    "        textAnnotations.append(tmp)\n",
+    "        \n",
+    "    return textAnnotations\n",
+    "\n",
+    "    if response.error.message:\n",
```

Comment:
This code cannot be reached. Put it above the `return`

Reply:
@marcenacp done